monai/apps/utils.py

```
@@ -121,6 +120,36 @@ def update_to(self, b: int = 1, bsize: int = 1, tsize: int | None = None) -> Non
+def safe_extract_member(member, extract_to):
+    """Securely verify compressed package member paths to prevent path traversal attacks"""
+    # Get member path (handle different compression formats)
+    if hasattr(member, 'filename'):
+        member_path = member.filename  # zipfile
+    elif hasattr(member, 'name'):
+        member_path = member.name  # tarfile
+    else:
+        member_path = str(member)
+
+    if hasattr(member, 'issym') and member.issym():
+        raise ValueError(f"Symbolic link detected in archive: {member_path}")
```

Ruff (0.12.2)

Avoid specifying long messages outside the exception class

(TRY003)